File: S1-voluntario/CarreraCarretera.rb
require_relative "Carrera.rb"


module S1
	public
	class CarreraCarretera < Carrera

		# def clone
		# 	CarreraCarretera.new
		# 	#new(valor)
		# end

		def initialize
			super
			@percent = 0.1
		end

		def getTipo
			"Carretera"
		end

		def setBici(bici)
			if bici.getTipo == "Montaña"
				puts "No se puede inscribir una bicicleta de Carretera a una carrera de Montaña"
			else
				@bicis << bici
			end
		end

		public :clone
	end
end

File: P1/S1/CarreraCarretera.java
package S1;

public class CarreraCarretera extends Carrera {
    CarreraCarretera()
    {
        super();
        percent = 0.1;
    }

    @Override
    void setBici(Bicicleta bici) {
        if(bici.getTipo() == "Montaña")
        {
            System.out.println("No se puede inscribir una bicicleta de Montaña a una carrera de Carretera");
            System.exit(1);
        }
        else
            bicis.add(bici);
    }

    @Override
    String getTipo() {
        return "Carretera";
    }
}